src/nodemanagerbase.cpp

```diff
@@ -19,54 +19,50 @@
  *  along with Quasar.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <algorithm>
+#include <stdexcept>
 
 #include <nodemanagerbase.h>
-#include <boost/foreach.hpp>
-#include <iostream>
 #include <opcua_basedatavariabletype.h>
-#include <stdexcept>
 #include <uadatavariablecache.h>
 
 static UaLocalizedText emptyDescription( "en_US", "" );
 
-NodeManagerBase::NodeManagerBase( const char* uri, bool sth, int hashtablesize ):
-    m_server(0),
-    m_nameSpaceUri(uri)
+NodeManagerBase::NodeManagerBase(
+        const char* sNamespaceUri,
+        bool        firesEvents,
+        OpcUa_Int32 nHashTableSize):
+    m_server(nullptr),
+    m_nameSpaceUri(sNamespaceUri)
 {
 
 }
 
 NodeManagerBase::~NodeManagerBase()
 {
-
-
-    std::cout << __FUNCTION__ << std::endl;
-    std::cout << "m_listNodes.size=" << m_listNodes.size() << std::endl;
     while (!m_listNodes.empty())
     {
         auto node = m_listNodes.back();
         m_listNodes.pop_back();
-
         delete node;
     }
-
 }
 
 UaNode* NodeManagerBase::getNode( const UaNodeId& nodeId ) const
 {
-    //TODO: the code belove is probably shitty - shall be decided one and forever whether getNode shall be const or not ...
     if (nodeId == m_serverRootNode.nodeId())
         return (UaNode*)(&m_serverRootNode);
 
-    BOOST_FOREACH (UaNode* node,  m_listNodes)
-    {
-        if (node->nodeId() == nodeId)
-            return node;
-    }
-    return 0; // not found
+    auto iterator = std::find_if(
+            m_listNodes.begin(),
+            m_listNodes.end(),
+            [nodeId](UaNode* node){return node->nodeId() == nodeId;});
+    if (iterator != m_listNodes.end())
```

Comment:
It's maybe a bit C++ golf-ish but I often use
`return (iterator != m_listNodes.end()) ? *iterator : nullptr`
just as an alternative

Reply:
OK, Will do!